Simplify mod point

mdharm opened the pull request:
Change mod point from using a struct Point with named fields to a more
simple tuple.  Also move the definition of ZERO to be an associated
const of struct Point -- this is useful in case we have ZERO values from
another module someday.

Also added unit tests for mod point.  All tests currently pass.  Due to the
nature of floating-point numbers, some tests are validated to within a
specified EPSILON value (also an associated const of struct Point).

mdharm committed:
Simplify mod point

Change mod point from using a struct Point with named fields to a more
simple tuple.  Also move the definition of ZERO to be an associated
const of struct Point -- this is useful in case we have ZERO values from
another module someday.

battjt commented on src/point.rs:
```diff
@@ -2,51 +2,82 @@ use rand::Rng;
-    }
-    pub fn magnitude(self) -> Float {
-        return self.magnitude_squared().sqrt();
+        self.0 * self.0 + self.1 * self.1
```
wow that reads weird.

mdharm replied:
Weird, but compact and (as far as I can tell) properly idiomatic.

I do see that I seem to have accidentally deleted the magnitude() method (I was manually merging files from another clone into this fork).  If you accept this, then I can provide a further patch and pull request to restore the magnitude method.

mdharm replied:
Actually, I just figured out how to amend a PR.  Added another commit to restore (and test) the magnitude() method.

mdharm committed:
Restore Point::magnitude() method

The magnitude() method got lost in a manual merge.  Restore it.

mdharm committed:
Merge branch 'main' into point_simplify

Pick up fixes to update_with() from branch simple_fixes that have been
merged into main.

mdharm committed:
Improve tests for update_with()

Add to the test cases for update_with() to verify that the mass not only
moves based on the current velocity, but that the velocity also updates
with applied force.

Starting with a mass at ZERO with velocity (1, 1), we check that after a
single time-step with no force, the mass has moved to (1, 1).  Checks
are performed both via PartialEq testing as well as via EPSILON.

Then apply force (2, 3) and verify that the mass has moved to (4, 5) --
since it was at (1, 1) and the velocity went from (1, 1) to (3, 4) due
to the force, the proper final position is (4, 5).

This could be further enahanced to actually test with different masses
(other than 1.0) and also to directly verify the calculated velocity
(rather than inferring it is correct from the position).
